### PR TITLE
Feature/nav bar

### PR DESCRIPTION
**Pull request by peterpep**
navbar

### CODE REVIEW

File: src/components/LogIn.vue [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    export default {
+        name: 'LogIn',
+        data() {
+            return {
+                user: {}
+            }
+        },
+        methods: {
+            close: function() {
+                this.$emit('close', 'stayOnPage');
```
Comment (bry-guy) [inline src/components/LogIn.vue]: this.$emit('clickedOutside');

File: src/App.vue [hunk 2]
```diff
@@ -1,13 +1,47 @@
...
+        this.logState = true;
+        this.showLogForm = false;
+      } else {
+        this.logOutState();
+      }
+    },
+    logOutState: function(e) {
+      this.logState = false;
+      this.showLogForm = false;
+      if (e != "stayOnPage")
```
Comment (bryansmi) [inline src/App.vue]: if (e == "RedirectToHome")

File: src/App.vue [hunk 3]
```diff
@@ -1,13 +1,47 @@
...
+    authUser: function(user) {
+      //mock api verification
+      if (user.username == "asd") {
+        this.logState = true;
+        this.showLogForm = false;
+      } else {
+        this.logOutState();
+      }
+    },
+    logOutState: function(e) {
```
Comment (bryansmi) [inline src/App.vue]: HandleClickOutside

Comment (bry-guy) [inline src/App.vue]: HandleLogOutButtonClicked: function() {
  //error handling
  LogOut('/');
}
 
HandleClickOutside_Login: function(e) {
  //error handling
  LogOut();
}
 
LogOut: function(redirectRoute) {
   this.logState = false;
   this.showLogForm = false;
   if (redirectRoute != null)
     Redirect(redirectRoute);
}

Redirect: function(route) {
  this.$router.push(route);
}

File: src/App.vue [hunk 4]
```diff
@@ -1,13 +1,58 @@
 <template>
   <div id="app">
-    <img src="./assets/logo.png">
-    <router-view></router-view>
+    <LogIn @clickedOutside="handleClickOutside_Login" class='loginOffset' v-if="this.showLogForm" @SubmitLog="authUser"></LogIn>
```
Comment (bry-guy) [inline src/App.vue]: LogIn renamed to LoginContainer

File: src/App.vue [hunk 5]
```diff
@@ -1,13 +1,58 @@
...
+    return {
+      logState: false,
+      showLogForm: false
+    }
+  },
+  methods: {
+    display: function(state) {
+      this.showLogForm = state;
+    },
+    authUser: function(user) {
```
Comment (bry-guy) [inline src/App.vue]: rename function to logIn

File: src/App.vue [hunk 6]
```diff
@@ -1,13 +1,58 @@
...
+        this.Redirect(e);
+      
+    },
+    handleLogOutButtonClicked: function() {
+      this.LogOut('/');
+    },
+    handleClickOutside_Login: function() {
+      this.LogOut();
+    },
+    Redirect: function(route) {
```
Comment (bry-guy) [inline src/App.vue]: Redirect renamed to redirect

File: src/App.vue [hunk 7]
```diff
@@ -1,13 +1,58 @@
...
+    authUser: function(user) {
+      //mock api verification
+      if (user.username == "asd") {
+        this.logState = true;
+        this.showLogForm = false;
+      } else {
+        this.logOutState();
+      }
+    },
+    LogOut: function(e) {
```
Comment (bry-guy) [inline src/App.vue]: LogOut renamed to logOut

File: src/components/NavBar.vue [hunk 8]
```diff
@@ -0,0 +1,110 @@
+<template>
+<div v-bind:class="{'modal-mask': (this.isFormActive)}">
+    <nav >
+        <ul><li><span>FLOWBASE</span></li></ul>
+        <ul class="navLinks">
+            <li><router-link to="/">Overview</router-link></li>
```
Comment (bry-guy) [inline src/components/NavBar.vue]: Add comment regarding nested/wrapped li's needing to be changed to class-targeted css

File: src/App.vue [hunk 9]
```diff
@@ -1,13 +1,58 @@
...
+    LogInContainer
+  },
+  data() {
+    return {
+      isUserLoggedIn: false,
+      showLogForm: false
+    }
+  },
+  methods: {
+    display: function(state) {
```
Comment (bry-guy) [inline src/App.vue]: change display to showLoginContainer

File: src/App.vue [hunk 10]
```diff
@@ -1,13 +1,58 @@
...
+      this.showLogForm = false;
+      if (e != null)
+        this.redirect(e);
+      
+    },
+    handleLogOutButtonClicked: function() {
+      this.logOut('/');
+    },
+    handleClickOutside_Login: function() {
+      this.logOut();
```
Comment (bry-guy) [inline src/App.vue]: this.showLoginContainer(false);

**Review by bry-guy — APPROVED** [review]
DOPE

**Pull request merged**